server/src/logic/expense_operations.rs

```diff
@@ -82,3 +61,43 @@ pub async fn find_predefined_expenses(
     let predefined_expenses = PredefinedExpense::find().all(conn).await?;
     Ok(predefined_expenses)
 }
+
+pub async fn create_predefined_expense(
+    conn: &DatabaseConnection,
+    req: NewPredefinedExpenseRequest,
+) -> Result<Id, ExpenseError> {
+    validate_recurrence_and_currency_types(conn, req.currency_type_id, req.recurrence_type_id)
+        .await?;
+    let predefined_expense = predefined_expense::ActiveModel {
+        id: NotSet,
+        name: Set(req.name),
+        description: Set(req.description),
+        value: Set(req.value),
+        currency_type_id: Set(req.currency_type_id),
+        recurrence_type_id: Set(req.recurrence_type_id),
+    };
+    let predefined_expense = predefined_expense.insert(conn).await?;
+    Ok(predefined_expense.id)
+}
+
+async fn validate_recurrence_and_currency_types(
+    conn: &DatabaseConnection,
+    currency_type_id: Id,
+    recurrence_type_id: Id,
+) -> Result<(), ExpenseError> {
+    let (referred_recurrence_type, referred_currency_type) = tokio::join!(
+        RecurrenceType::find()
+            .filter(recurrence_type::Column::Id.eq(recurrence_type_id))
+            .one(conn),
+        CurrencyType::find()
+            .filter(currency_type::Column::Id.eq(currency_type_id))
+            .one(conn)
+    );
+    if referred_recurrence_type?.is_some() && referred_currency_type?.is_some() {
+        Ok(())
```

Comment:
I think this is less readable, than the code you extracted.
But you recommended once before, so I think you like it this way.
Its okayish for me

Reply:
My reasoning is that usually the requests will come from our front-end and those request - if there is no bugs - should contain valid ids. And smart people on the internet said to me that in if statements the more likely outcome should come first because compilation and cpu reasons. More info [here](https://stackoverflow.com/a/46835888). I would love to hear your opinion on it.

Reply:
Okay, this small if does not make a huge difference in any case and readibility is more of a concern for us than performance, this little webserver is already plenty performant. So I will flip the if.
Defensive programming is cool and we should adhere to it. 🚀 Be on the lookout for nonadherence in my code when reviewing. Altough I think there are some cases such as slow if test clauses where we can stray a bit so likely outcomes come first, expensive to test outcomes come last.

Other parts of the function are cool right?